File: test_server.py
import requests
import base64
import os

# The URL of the live server
SERVER_URL = 'https://alttextserver.symm.app/generate-alt-text'

# A small, 1x1 black pixel JPEG image as a base64 string
base64_image = '/9j/4AAQSkZJRgABAQEASABIAAD/2wBDAAMCAgICAgMCAgIDAwMDBAYEBAQEBAgGBgUGCQgKCgkICQkKDA8MCgsOCwkJDRENDg8QEBEQCgwSExIQEw8QEBD/2wBDAQMDAwQDBAgEBAgQCwkLEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBD/wAARCAABAAEDASIAAhEBAxEB/8QAFQABAQAAAAAAAAAAAAAAAAAAAAn/xAAUEAEAAAAAAAAAAAAAAAAAAAAA/8QAFAEBAAAAAAAAAAAAAAAAAAAAAP/EABQRAQAAAAAAAAAAAAAAAAAAAAD/2gAMAwEAAhEDEQA/AL+AAf/Z'

def test_alt_text_server():
    print(f"Testing alt text server at: {SERVER_URL}")

    try:
        request_body = {
            'mimeType': 'image/jpeg',
            'base64Data': base64_image,
        }

        print('Sending request to server...')
        response = requests.post(SERVER_URL, json=request_body, headers={
            'Content-Type': 'application/json',
            'Origin': 'https://alttext.symm.app'  # A valid origin
        })

        print(f"Response Status: {response.status_code}")
        response_data = response.json()

        if not response.ok:
            print('Test Failed: Server returned an error.')
            print(f'Response: {response_data}')
            exit(1)

        if response_data.get('altText') and len(response_data['altText']) > 0:
            print('Test Passed: Server returned alt text!')
            print(f"Generated Alt Text: {response_data['altText']}")
        else:
            print('Test Failed: Server did not return alt text.')
            print(f'Response: {response_data}')
            exit(1)

    except Exception as e:
        print('Test Failed: An error occurred during the test.')
        print(e)
        exit(1)

if __name__ == "__main__":
    test_alt_text_server() 

File: test_server.js
const fetch = require('node-fetch');
const fs = require('fs');
const path = require('path');

// The URL of the live server
const SERVER_URL = 'https://alttextserver.symm.app/generate-alt-text';

// A small, sample image file to send for testing
const SAMPLE_IMAGE_PATH = path.join(__dirname, 'sample-image.jpg');

// A small, 1x1 black pixel JPEG image as a base64 string
const base64Image = '/9j/4AAQSkZJRgABAQEASABIAAD/2wBDAAMCAgICAgMCAgIDAwMDBAYEBAQEBAgGBgUGCQgKCgkICQkKDA8MCgsOCwkJDRENDg8QEBEQCgwSExIQEw8QEBD/2wBDAQMDAwQDBAgEBAgQCwkLEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBD/wAARCAABAAEDASIAAhEBAxEB/8QAFQABAQAAAAAAAAAAAAAAAAAAAAn/xAAUEAEAAAAAAAAAAAAAAAAAAAAA/8QAFAEBAAAAAAAAAAAAAAAAAAAAAP/EABQRAQAAAAAAAAAAAAAAAAAAAAD/2gAMAwEAAhEDEQA/AL+AAf/Z';

// Function to create a dummy image file for testing
function createDummyImage() {
  const imageBuffer = Buffer.from(base64Image, 'base64');
  fs.writeFileSync(SAMPLE_IMAGE_PATH, imageBuffer);
}

async function testAltTextServer() {
  console.log('Creating dummy image file...');
  createDummyImage();

  console.log(`Testing alt text server at: ${SERVER_URL}`);

  try {
    const imageData = fs.readFileSync(SAMPLE_IMAGE_PATH);
    const base64Data = imageData.toString('base64');

    const requestBody = {
      mimeType: 'image/jpeg',
      base64Data: base64Data,
    };

    console.log('Sending request to server...');
    const response = await fetch(SERVER_URL, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Origin': 'https://alttext.symm.app' // A valid origin
      },
      body: JSON.stringify(requestBody),
    });

    console.log(`Response Status: ${response.status}`);
    const responseData = await response.json();

    if (!response.ok) {
      console.error('Test Failed: Server returned an error.');
      console.error('Response:', responseData);
      process.exit(1);
    }

    if (responseData.altText && responseData.altText.length > 0) {
      console.log('Test Passed: Server returned alt text!');
      console.log('Generated Alt Text:', responseData.altText);
    } else {
      console.error('Test Failed: Server did not return alt text.');
      console.error('Response:', responseData);
      process.exit(1);
    }

  } catch (error) {
    console.error('Test Failed: An error occurred during the test.');
    console.error(error);
    process.exit(1);
  } finally {
    // Clean up the dummy image file
    if (fs.existsSync(SAMPLE_IMAGE_PATH)) {
      fs.unlinkSync(SAMPLE_IMAGE_PATH);
    }
  }
}

testAltTextServer(); 